Source/Factories/JSONFactory.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  JSONFactory.swift
+//  Matrioska
+//
+//  Created by Mathias Aichinger on 11/01/2017.
+//  Copyright © 2017 runtastic. All rights reserved.
+//
+
+import Foundation
+
+/// An error type for JSONFactory
+///
+/// - missing: specifies that a mandatory key is missing
+enum JSONFactoryError: Error {
+    case missing(JSONObject, String)
+}
+
+/// A factory that wraps ComponentFactory objects and uses them to produce Components
+public final class JSONFactory {
+    
+    static let typeKey = "type"
+    static let metaKey = "meta"
+    static let childrenKey = "children"
```

Comment:
Don't agree with using children instead of child. If we want to have wrappers (and I think we want) they shouldn't be an hacky cluster (a cluster that ignore children other than the first.
Please couldn't you guys reconsider this decision or try to explain why you think it should be like this?